docs/userGuide/syntax/code.mbdf

```diff
@@ -47,9 +61,111 @@ As seen above, line numbers are automatically provided for a code block. To hide
 </span>
 </include>
 
-<small>More info: https://www.markdownguide.org/extended-syntax#fenced-code-blocks</small>
+##### Line highlighting
+To highlight lines, add the attribute `highlight-lines` with the line numbers as value, as shown below. You can specify ranges or individual line numbers.
+
+<include src="outputBox.md" boilerplate >
+<span id="code">
+
+```` {.no-line-numbers}
+```java {highlight-lines="2,4,6-8"}
+import java.util.List;
+
+public class Inventory {
+    private List<Item> items;
+
+    public int getItemCount(){
+        return items.size();
+    }
+
+    public void generateBill(Billable b){
+        // ...
+    }
+
+    public void add(Item s) {
+        items.add(s);
+    }
+}
+```
+````
+</span>
+<span id="output">
+
+```java {highlight-lines="2,4,6-8"}
+import java.util.List;
+
+public class Inventory {
+    private List<Item> items;
+
+    public int getItemCount(){
+        return items.size();
+    }
+
+    public void generateBill(Billable b){
+        // ...
+    }
+
+    public void add(Item s) {
+        items.add(s);
+    }
+}
+```
+</span>
+</include>
+
+##### Headers
+To add headers, add the attribute `header` with the header text as the value, as shown below.
+
+<include src="outputBox.md" boilerplate >
+<span id="code">
+
+```` {.no-line-numbers}
+```xml {heading="Heading title"}
+<foo>
+  <bar type="name">goo</bar>
+</foo>
+```
+````
+</span>
+<span id="output">
+
+```xml {heading="Heading title"}
+<foo>
+  <bar type="name">goo</bar>
+</foo>
+```
+</span>
+</include>
+
+##### Using them altogether
```

Comment:
I feel that this section isn't very necessary, since combining selectors should be fairly self explanatory. I'd lean towards removing this section to avoid bloating the user guide.

Reply:
Turns out we do need it 😅